lib/beacon_api/controllers/v1/beacon_controller.ex

```diff
@@ -1,11 +1,22 @@
 defmodule BeaconApi.V1.BeaconController do
+  alias BeaconApi.ApiSpec
   alias BeaconApi.ErrorController
+
   alias LambdaEthereumConsensus.ForkChoice
   alias LambdaEthereumConsensus.Store.BlockStore
   use BeaconApi, :controller
 
+  plug(OpenApiSpex.Plug.CastAndValidate, json_render_error_v2: true)
+
+  def open_api_operation(action) do
+    apply(__MODULE__, :"#{action}_operation", [])
+  end
```

Comment:
What's `action`? Is it an atom? Is there a bounded amount of `action`s?

Reply:
an `action` is an atom that is the MVC controller action's function name. In the Beacon API case, the actions are  `:get_state_root`, `:get_block_root`,  `:get_block`

https://github.com/open-api-spex/open_api_spex/blob/master/examples/phoenix_app/lib/phoenix_app_web/controllers/user_controller_with_struct_specs.ex#L17